test/with_dummyserver/test_socketlevel.py

```diff
@@ -837,9 +856,9 @@ def socket_handler(listener):
                        ':9A:8C:B6:07:CA:58:EE:74:5E')
 
         def request():
+            pool = HTTPSConnectionPool(self.host, self.port,
```

Comment:
This is done because there's a chance that pool is not defined by the time we get outside of the `try-catch` block (eg If an exception occurs during the creation of the `HTTPSConnectionPool`). I can revert this change if that's what's wanted.